lib/ansible/modules/remote_management/oneview/oneview_datacenter.py

```diff
@@ -0,0 +1,174 @@
+              C(absent) will remove the resource from OneView, if it exists.
+              C(present) will ensure the datacenter resource exists and its properties are compliant with HPE OneView.
+        choices: ['present', 'absent']
+        default: present
+    data:
+        description:
+            - List with Datacenter properties and its associated states.
+        required: true
+
+extends_documentation_fragment:
+    - oneview
+    - oneview.validateetag
+'''
+
+EXAMPLES = '''
+- name: Add a Datacenter
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        contents:
+            # You can choose either resourceName or resourceUri to inform the Rack
+            - resourceName: '{{ datacenter_content_rack_name }}' # option 1
+              resourceUri: ''                                    # option 2
+              x: 1000
+              y: 1000
+        depth: 6000
+        name: "MyDatacenter"
+        width: 5000
+  no_log: true
+  delegate_to: localhost
+
+- name: Update the Datacenter with specified properties (no racks)
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        contents: ~
+        coolingCapacity: '5'
+        coolingMultiplier: '1.5'
+        costPerKilowattHour: '0.10'
+        currency: USD
+        defaultPowerLineVoltage: '220'
+        deratingPercentage: '20.0'
+        depth: 5000
+        deratingType: NaJp
+        name: "MyDatacenter"
+        width: 4000
+  no_log: true
+  delegate_to: localhost
+
+- name: Rename the Datacenter
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        name: "MyDatacenter"
+        newName: "My Datacenter"
+  no_log: true
+  delegate_to: localhost
+
+- name: Remove the Datacenter
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: absent
+    data:
+        name: 'My Datacenter'
+  no_log: true
+  delegate_to: localhost
+'''
+
+RETURN = '''
+datacenter:
+    description: Has the OneView facts about the Datacenter.
+    returned: On state 'present'. Can be null.
+    type: dict
+'''
+
+from ansible.module_utils.oneview import OneViewModuleBase, OneViewModuleResourceNotFound
+
+
+class DatacenterModule(OneViewModuleBase):
+    MSG_CREATED = 'Datacenter added successfully.'
+    MSG_UPDATED = 'Datacenter updated successfully.'
+    MSG_ALREADY_PRESENT = 'Datacenter is already present.'
+    MSG_DELETED = 'Datacenter removed successfully.'
+    MSG_ALREADY_ABSENT = 'Datacenter is already absent.'
+    MSG_RACK_NOT_FOUND = 'Rack was not found.'
+    RESOURCE_FACT_NAME = 'datacenter'
+
+    argument_spec = dict(
```

Comment:
Can you please ensure the module works with checkmode

Reply:
The ```check_mode``` is disabled for all oneview modules. Please see this review. https://github.com/ansible/ansible/pull/32701#discussion_r150933457